### PR TITLE
Add fresnels

### LINKED ISSUE #136: RFC: Evaluate S(x) part of Fresnel integral (fresnels)
## Checklist

> Please ensure the following tasks are completed before filing an issue.

-   [x] Read and understood the [Code of Conduct][code-of-conduct].
-   [x] Searched for existing issues and pull requests.
-   [x] If this is a general question, searched the [FAQ][faq] for an existing answer.
-   [x] If this is a feature request, the issue name begins with `RFC:`.

## Description

Separate fast-path to evaluate *only* S(x) part of Fresnel integral.

## Related Issues

> Does this issue (or feature request) have any related issues?

See #134 for more info.

## Questions

> Any questions for reviewers?

No.

## Other

No.

[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md

[faq]: https://github.com/stdlib-js/stdlib/blob/develop/FAQ.md

### PR DESCRIPTION
**Pull request by rreusser**
Resolves #136 .

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing], including the relevant style guides.
-   [x] Read and understand the [Code of Conduct][code-of-conduct].
-   [x] Read and understood the [licensing terms][license].
-   [x] Searched for existing issues and pull requests **before** submitting this pull request.
-   [x] Filed an issue (or an issue already existed) **prior to** submitting this pull request.
-   [x] Rebased onto latest `develop`.
-   [x] Submitted against `develop` branch.

## Description

> What is the purpose of this pull request?

This PR adds `fresnels` for computing the S(x) term of the Fresnel integral.

## Related Issues

> Does this pull request have any related issues?

See also: #135.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

`fresnel` could use both this and `fresnelc` for computing the separate expansions for small x. There's no overlap so that it would reduce duplication of the code, though that's a refactoring that would require the separate versions to be merged first.

It's also possible that the large-x expansion could probably be streamlined to more efficiently compute the separate expansions, though I think that's secondary to an effective, accurate computation.

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

[code-of-conduct]: https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md

[license]: https://github.com/stdlib-js/stdlib/blob/develop/LICENSE

### CODE REVIEW

**Comment by kgryte** [comment]
@rreusser If there is no overlap or perf gains to be attained by computing, e.g., simultaneously, I am fine with merging the separate implementations prior to a combined API.

**Comment by rreusser** [comment]
There is overlap, but only for some ranges. Small numbers use separate expansions. Large numbers use a coupled expansion that makes it natural to compute both together. There's probably a large number expansion that is slightly more efficient by only using `sine` *or* `cosine` (not both for the sake of the phase change), but I don't know it.

**Comment by kgryte** [comment]
Up to you. In general, I am fine with code duplication where it makes sense. Use your best judgment.

**Comment by rreusser** [comment]
I think the small amount of duplication makes it more reasonable to just have all three entirely separate. Really getting it _perfect_ would require additional high-precision re-derivation.

**Comment by kgryte** [comment]
Sounds good to me. :)

File: lib/node_modules/@stdlib/math/base/special/fresnels/lib/fresnels.js [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+var sincos = require( '@stdlib/math/base/special/sincos' );
+var abs = require( '@stdlib/math/base/special/abs' );
+var evalrational = require( '@stdlib/math/base/tools/evalrational' ).factory;
+var HALF_PI = require( '@stdlib/math/constants/float64-half-pi' );
+var PI = require( '@stdlib/math/constants/float64-pi' );
+
+
+// VARIABLES //
+
+/* S(x) for small x */
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/fresnels/lib/fresnels.js]: Comment style.

File: lib/node_modules/@stdlib/math/base/special/fresnels/lib/fresnels.js [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+	6.07366389490084639049E11,
+	2.24411795645340920940E10,
+	4.19320245898111231129E8,
+	5.17343888770096400730E6,
+	4.55847810806532581675E4,
+	2.81376268889994315696E2,
+	1.00000000000000000000E0
+];
+
+/* Auxiliary function f(x) */
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/fresnels/lib/fresnels.js]: Comment style.

File: lib/node_modules/@stdlib/math/base/special/fresnels/lib/fresnels.js [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+	1.12699224763999035261E-8,
+	1.84627567348930545870E-6,
+	1.55934409164153020873E-4,
+	6.44051526508858611005E-3,
+	1.16888925859191382142E-1,
+	7.51586398353378947175E-1,
+	1.00000000000000000000E0
+];
+
+/* Auxiliary function g(x) */
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/fresnels/lib/fresnels.js]: Comment style.

**Review by kgryte — CHANGES_REQUESTED** [review]

File: lib/node_modules/@stdlib/math/base/special/fresnels/lib/fresnels.js (hunk 1 above)
Comment (rreusser) [reply lib/node_modules/@stdlib/math/base/special/fresnels/lib/fresnels.js]: ✅

File: lib/node_modules/@stdlib/math/base/special/fresnels/lib/fresnels.js (hunk 2 above)
Comment (rreusser) [reply lib/node_modules/@stdlib/math/base/special/fresnels/lib/fresnels.js]: ✅

File: lib/node_modules/@stdlib/math/base/special/fresnels/lib/fresnels.js (hunk 3 above)
Comment (rreusser) [reply lib/node_modules/@stdlib/math/base/special/fresnels/lib/fresnels.js]: ✅

**Review by kgryte — APPROVED** [review]

**Comment by kgryte** [comment]
@rreusser Thanks for your work on this! Will merge once CircleCI finishes. :)

**Pull request merged**